logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/auth/LoginUseCaseTest.kt

```diff
@@ -123,7 +123,43 @@ class LoginUseCaseTest {
         }
 
     @Test
-    fun `given LoginUseCase is invoked, When ValidateEmailUseCase return false and validateUserHandleUseCase loginWithEmail and storeSession return success, then returns success`() =
+    fun givenLoginUseCaseIsInvoked_shouldStoreSessionReturnsFalse_whenValidateEmailUseCase_loginWithEmailAndStoreSessionReturnSuccess_thenDoNotStoreSessionAndReturnSuccess() =
+        runTest {
+            given(validateEmailUseCase)
+                .invocation { invoke(TEST_EMAIL) }
+                .then { true }
+            given(validateUserHandleUseCase)
+                .invocation { invoke(TEST_EMAIL) }
+                .then { false }
+            given(loginRepository)
+                .coroutine { loginWithEmail(TEST_EMAIL, TEST_PASSWORD, TEST_PERSIST_CLIENT, TEST_SERVER_CONFIG) }
+                .then { Either.Right(TEST_AUTH_SESSION) }
+
+            val loginUserCaseResult = loginUseCase(TEST_EMAIL, TEST_PASSWORD, TEST_PERSIST_CLIENT, TEST_SERVER_CONFIG) { false }
+
+            assertEquals(loginUserCaseResult, AuthenticationResult.Success(TEST_AUTH_SESSION))
+            verify(validateEmailUseCase)
+                .invocation { invoke(TEST_EMAIL) }
+                .wasInvoked(exactly = once)
+            verify(validateUserHandleUseCase)
+                .function(validateUserHandleUseCase::invoke).with(any())
+                .wasNotInvoked()
+            verify(loginRepository)
+                .coroutine { loginWithEmail(TEST_EMAIL, TEST_PASSWORD, TEST_PERSIST_CLIENT, TEST_SERVER_CONFIG) }
+                .wasInvoked(exactly = once)
+            verify(loginRepository)
+                .suspendFunction(loginRepository::loginWithHandle).with(any(), any(), any(), any())
+                .wasNotInvoked()
```

Comment:
```suggestion
```
Considering that this test seems to aim on testing how `shouldStoreSession` affects the UseCase, we can simplify and take some assertions out.

Reply:
Sure, I removed unnecessary ones.